src/main/java/Deadline.java

```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+    private String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```

Comment:
Perhaps you would like to use `String.format()` instead of concatenation to rewrite this part to make it neater and easier to make changes in the future?

```suggestion
        return String.format("[D]%s (by %s )", super.toString(), by);
```